crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -312,6 +312,44 @@ def create_proposal
     base
   end
 
+  # Simple check if HA clusters report some problems
+  # If there are no problems, empty hash is returned.
+  # If this fails, information about failed actions for each cluster founder is
+  # returned in a hash that looks like this:
+  # {
+  #     "crm_failures" => {
+  #             "node1" => "reason for crm status failure"
+  #     },
+  #     "failed_actions" => {
+  #             "node2" => "Failed action on this node"
+  #     }
+  # }
+  # User has to manually clean pacemaker resources before proceeding with the upgrade.
+  def check_cluster_health
```

Comment:
Could we have that code in a different file? This really doesn't belong to the crowbar service :-) (should probably be in a lib)